uvicorn/protocols/websockets/wsproto_impl.py

```diff
@@ -129,10 +129,9 @@ def resume_writing(self):
         self.writable.set()
 
     def shutdown(self):
-        self.queue.put_nowait({"type": "websocket.disconnect", "code": 1012})
-        output = self.conn.send(wsproto.events.CloseConnection(code=1012))
-        self.transport.write(output)
```

Comment:
is this needed ? won't it leave the protocol in a state it should not be in, how it will know it's in a closing state ?

Reply:
It's same as the other implementation - `run_asgi` cleans up each time. The current flow just injects data into an active connection causing worse issues :(